metaseq_cli/train.py

```diff
@@ -486,6 +494,36 @@ def post_checkpoint_callback(cfg, filename):
             except (FileNotFoundError, AssertionError) as e:
                 logger.info(f"could not upload {filename}: {e}")
 
+        if do_evaluate:
+            _run_evaluations(
+                cfg.checkpoint.eval_command,
+                cfg.checkpoint.cloud_upload_path,
+                filename,
+                checkpoint_suffix,
+            )
+
+
+def _run_evaluations(eval_cmd, cloud_upload_path, local_file, checkpoint_suffix):
+    # WARNING: evals command needs to properly wait for
+    # all ranks to have finished uploading. Here we just
+    # wait for rank 0.
+    if distributed_utils.get_global_rank() != 0:
+        return
```

Comment:
I think we need to figure this out, not just pray on rank0 to be done. We'll need this for NFS checkpointing too.

See here https://fb.workplace.com/groups/319878845696681/posts/2360341237454404

Reply:
Lol exactly the question I had, I'll give it a try (gloo + monitored_barrier)

Reply:
Added gloo pg + monitored_barrier